packages/oui-datagrid/src/datagrid.controller.js

```diff
@@ -313,6 +318,37 @@ export default class DatagridController {
         };
     }
 
+    getSelectedRows () {
+        return this.selectedRows.reduce((result, selected, index) => {
+            if (selected) {
+                result.push(this.displayedRows[index]);
+            }
+            return result;
+        }, []);
+    }
+
+    onSelectRow (index, selected) {
+        const rowCount = this.displayedRows.length;
+        this.selectedRows[index] = selected;
+        const selectedRowsCount = this.getSelectedRows().length;
+
+        if (selectedRowsCount === rowCount) {
+            this.selectAllRows = true;
+        } else if (selectedRowsCount === 0) {
+            this.selectAllRows = false;
+        } else {
+            this.selectAllRows = null;
+        }
+    }
+
+    toggleSelectAllRows (modelValue) {
```

Comment:
`currentSelectionStatus` ?

Reply:
We could keep the same logic as above, `toggleRowsSelection`